source/_docs/installation/synology.markdown

```diff
@@ -2,78 +2,196 @@
 ```
 
+Edit the following 2 files.
 
-Check the path to python3 (assumed to be /volume1/@appstore/py3k/usr/local/bin)
+These edits will make cross compiled Python module packages, which Home Assistant needs to install. These edits also enable use of the "[Cloud](/components/cloud/)" and [Homekit](/components/homekit/)" components.
+Most components will setup just fine, but certain components need a python package that compiles on installation, which will fail as Synology did not provide a compiler, causing said need for cross compilation.
 
-```bash
-# cd /volume1/@appstore/py3k/usr/local/bin
+Edit "*~/spksrc/spk/python3/src/requirements.txt*" and add the following text to the end of the file:
 ```
+## It may happen you want to add a component to Home Assistant, but it fails to install a Python package.
+## In the logs you see a error "Unable to install package *Module_Name*", if you install that
+# package with pip, some of its required packages fail with error "Failed building wheel for *Module_Name*"
+## Add these Python modules/packages that fail with that "building wheel" error in to the list below.
 
-Install PIP (Python's package management system)
+# Example format of module:
+#pythonmodule==version
 
-```bash
-# ./python3 -m ensurepip
-```
+## Cross compilation requirements for installing Home Assistant (Tested to work on 84.3).
+cffi==1.11.5
+bcrypt==3.1.4
+cryptography==2.3.1
 
-Use PIP to install Homeassistant package 0.64.3
+## Cross compilation requirement for installing "Warrent" module (Needed by "Cloud" Component)
+pycryptodome==3.7.2
 
-```bash
-# ./python3 -m pip install homeassistant==0.64.3
+## Cross compilation requirement for installing "HAP_python" module (Needed by "Homekit" Component)
+curve25519-donna==1.3
+ed25519==1.4
 ```
 
-Create homeassistant config directory & switch to it
-
+Add these edits to fix OpenSSL errors when using the "[Xiaomi_Aqara](/components/Xiaomi_Aqara/)" component.  
+Edit "*~/spksrc/spk/python3/Makefile*" and add the following text above the text "**include ../../mk/spksrc.spk.mk**":
+```makefile
+## Makefile variable needed to fix "_openssl.so: undefined symbol: pthread_atfork" error when using "xiaomi_aqara" component.
+export CFLAGS=-pthread
+```
+#### {% linkable_title Compiling the Python 3 package %}
+Make the Python 3 package for your Synology model, please replace "arch-**XXXX**" by the apporiate architecture of your Synology. For a list of architectures, look at this [list of architectures](https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model) accepted by spksrc. Depending on your computer, compilation may take a hour or more (Significantly less if you have a SSD and a moderately good CPU).
 ```bash
-# mkdir /volume1/homeassistant
-# chown homeassistant /volume1/homeassistant 
-# chmod 755 /volume1/homeassistant
-# cd /volume1/homeassistant
+# sudo docker run -it -v ~/spksrc:/spksrc synocommunity/spksrc /bin/bash
```

Comment:
`sudo docker run -it -v ~/spksrc:/spksrc synocommunity/spksrc  -w /spksrc /bin/bash`

You can almost one-shot this.  Two command lines would be:

```
sudo docker run -it --rm -v ~/spksrc:/spksrc -w /spksrc synocommunity/spksrc  make setup
sudo docker run -it --rm -v ~/spksrc:/spksrc -w /spksrc synocommunity/spksrc  make -C spk/python3 arch-XXXX-6.1
```

Reply:
Although, might adding the version be better in the case a user runs a older version of DSM (e.g., 6.1)? I mean in the case his NAS is no longer supported and no new DSM is available?
Might new DSM become incompatible because of setting a DSM version?

Reply:
Every time the user runs `docker run`, his container will be removed.
In case the compilation is cancelled or fails, his docker container would also be removed?
Is it not better to let the user decide to remove the containers?

I might even have to add one anyway in case the user runs `docker pull`, but the download fails, filling "docker/overlay" with corrupted container
I tried looking for a command to delete corrupted containers, but "docker rm" did not seem to erase these. I was able to erase these by stopping docker service and deleting "/var/lib/docker", but what are the odds the user is actually using Docker and has to re-download all the Docker containers...?

Reply:
Never mind, tested again, seems the thing with the Docker containers were just a placebo effect and it cleans-up just fine.

Though I still oppose to removing the docker container when running the compilation command, a better solution is adding a separate command for removing the docker container, that way the user won't have to re-download the container on every run and it is easier to decide if he wants to remove the container and the user won't be confused if he runs the command a second time and gets a error because the docker container was removed.